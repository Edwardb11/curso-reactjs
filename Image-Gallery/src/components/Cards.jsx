import React, { useState, useEffect } from 'react'
import Card from './Card'

const Cards = () => {
const [images, setImages] = useState({})

const peticion = async () => {
const res= await fetch('https://api.unsplash.com/photos/random/?client_id=TlD4jwNraHpTsPCwcSYEHyYJNNf2pSnBnq3mgDEjmkQ')
const data = await res.json()
// console.log(data)
setImages(data.urls.regular)
}
useEffect(() => {
    peticion();
},[])

  return (
   <>
   <Card/>
   </>
  )
}

export default Cards