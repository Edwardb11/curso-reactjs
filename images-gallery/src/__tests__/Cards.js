import { render, screen } from "@testing-library/react";
import Cards from "../components/Cards";

describe("<Cards/>", () => {
  it("Snapshot cards", () => {
    render(<Cards />);
    expect(screen.getByLabelText("imagenes")).toMatchSnapshot();
  });
  it("Snapshot card",async () => {
    render(<Cards />);
    // console.log(await screen.findAllByLabelText("img"))
    const res = await screen.findAllByLabelText("img");

    expect(res).toMatchSnapshot();
  });
});
