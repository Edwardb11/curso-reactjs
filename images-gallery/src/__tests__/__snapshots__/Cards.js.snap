// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cards/> Snapshot cards 1`] = `
<div
  aria-label="imagenes"
  class="row"
/>
`;
